File: Composite/PHP/Group.php
<?php

require_once "IUser.php";

class Group implements IUser {
    private array $users;

    public function add(IUser $user) {
        $this->users[] = $user;
    }

    public function getUserDetail()
    {
        foreach ($this->users as $user) {
            $user->getUserDetail();
        }
    }
}

File: Composite/Golang/Group.go
package main

type Group struct {
	Users []IUser
}

func NewGroup() *Group {
	return &Group{
		Users: make([]IUser, 0),
	}
}

func (group *Group) add(user IUser) {
	group.Users = append(group.Users, user)
}

func (group *Group) getUserDetail() {
	for _, user := range group.Users {
		user.getUserDetail()
	}
}